suzieq/poller/controller/credential_loader/base_credential_loader.py

```diff
@@ -35,6 +35,10 @@ def __init__(self, init_data: Dict) -> None:
         self._conf_passphrase = None
         self._conf_keyfile = None
         self._conf_username = None
+        self._valid_fields = ['username', 'name', 'type',
+                              'password', 'ssh-passphrase', 'keyfile']
+
+        self._validate_config(init_data)
```

Comment:
Do we need a mandatory fields check i.e. for an auth are some fields mandatory such as name and type and if its missing, complain and abort?

Reply:
The mandatory field check is done in each different plugin. For the default auth credentials plugin, none of the fields is mandatory. For example we can use the native format in this way
```yaml
sources:
- name: native0
  hosts:
  - url: ssh://vagrant@1.1.1.1
  - url: ssh://vagrant@1.1.1.2
auths:
- name: auth0
  password: vagrant
namespaces:
- name: native-ns
  source: native0
  auth: auth0
```
If some mandatory fields are needed by a plugin, it is its duty to check them. For example check https://github.com/netenglabs/suzieq/blob/448b6d996a913b698d51bf1cb49671d3d6a7f5e8/suzieq/poller/controller/source/netbox.py#L76, netbox checks if `token` and `url` are present in the config.

This is not the best method to do the validation but it works. After the alpha, we were thinking about using JSONSchema fot validation

Reply:
Aren't name (and type?) mandatory for each plugin

Reply:
Think about the validation as 2 steps:
- the first step is to validate the inventory as is. Here we check if the structure of each object config is right. For example checking if the name is specified (type is not mandatory because both sources and auth has a default type)
- the second step is to validate each component. In order to do this, we need to create the objects and each one of them runs a validation before starting. Here we validate that the users doesn't put a value which is not used by object